app/uk/gov/hmrc/economiccrimelevyregistration/controllers/deregister/DeregisterCheckYourAnswersController.scala

```diff
@@ -73,8 +72,27 @@ class DeregisterCheckYourAnswersController @Inject() (
     )
   }
 
-  def onSubmit(): Action[AnyContent] = authorise { _ =>
-    Redirect(routes.DeregistrationRequestedController.onPageLoad())
+  def onSubmit(): Action[AnyContent] = (authorise andThen getDeregistrationData).async { implicit request =>
+    val deregistration = request.deregistration
+    (for {
+      eclReference <- valueOrError(request.eclRegistrationReference, "ECL reference")
+      subscription <- eclRegistrationService.getSubscription(eclReference).asResponseError
+      address       = subscription.correspondenceAddressDetails
+      name         <- valueOrError(deregistration.contactDetails.name, "Name")
+      email        <- valueOrError(deregistration.contactDetails.emailAddress, "Email address")
+    } yield (email, name, eclReference, address)).fold(
+      err => routeError(err),
+      data => {
+        emailService.sendDeregistrationEmail(
```

Comment:
This future is not awaited, so you are essentially firing and forgetting. Lift it into the for comprehension. The redirect can stay here though.

Reply:
You should also write some tests to handle the email failures and all those valueOrError cases up there.